### PR TITLE
Feat/agreement creation flow

### PR DESCRIPTION
**Pull request by mehtaphysical**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 1
+---
+
+# Architecture
+
+## Modules
+
+zkSig is comprised of the following modules:
```
Comment (miniponz) [inline docs/index.md]:
```suggestion
zkSig comprises the following modules:
```

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+- **Authentication** - user provides credentials to prove who they are
+- **Encrypted Storage** - used to store encrypted files containing details about agreements and signatures
+- **Public Audit Trail** - a chronological record of agreement events stored in public as hashes
+- **Event Broadcaster** - listens to agreement events and broadcasts messages to smart contracts and webhooks
```
Comment (miniponz) [inline docs/index.md]: Struggling with the way these are worded a little. For consistency maybe:
- **Authentication** - handles credentials to prove who users are
- **Encrypted Storage** - stores encrypted files containing details about agreements and signatures
- **Public Audit Trail** - publicly stores a chronological record of agreement events as hashes
- **Event Broadcaster** - listens to agreement events and broadcasts messages to smart contracts and webhooks

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+Because of the modular design, zkSig can scale into new jurisdictions and new use cases by implementing
+different versions of the above modules. For example, currently we use [Web3Auth](https://web3auth.io/)
+to handle authentication. However, in the future we can implement new authentication modules like Decentralized
+Identity or Centralized Public Key.
+
+## Features & Flows
+
+The above modules are composed in different to create different features and user flows.
```
Comment (miniponz) [inline docs/index.md]: 1) wut?

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+```mermaid
+sequenceDiagram
+  actor Recipient
+  participant PublicAuditTrail
+  participant ZeroKnowledge
+
+  Recipient ->> PublicAuditTrail: Check hashes stored on public audit trail
+  Recipient ->> ZeroKnowledge: Call on-chain verification method with provided ZKP
+  ZeroKnowledge -->> Recipient: Verification result
+```
```
Comment (miniponz) [inline docs/index.md]: This one is a weird scale compared to the others. nbd, but nice to change in the future

File: docs/agreement-creation/agreement-processing.md [hunk 5]
```diff
@@ -18,18 +16,73 @@ An agreement's PDF is processed with the following goals:
 - have the ability to prove that a page is contained within a PDF without sharing the original PDF
 - have the ability to prove that a form field is contained within a PDF without sharing the original PDF
 
-In order to accomplish this we create two sparse merkle trees (SMTs): an SMT that represents the pages of a PDF
+To accomplish this two sparse merkle trees (SMTs) are created: an SMT that represents the pages of a PDF
```
Comment (miniponz) [inline docs/agreement-creation/agreement-processing.md]: Worth linking `sparse merkle tree` to something like https://wiki.polygon.technology/docs/zkEVM/zkProver/sparse-merkle-tree/ ?

**Review by miniponz — COMMENTED** [review]
Overall comment: we need to decide on `zksig` vs `zkSig` and then stick with it.

**Comment by adrimonasterios** [comment]
> Overall comment: we need to decide on `zksig` vs `zkSig` and then stick with it.

Yeap! Totally

**Review by miniponz — COMMENTED** [review]
yes i like it now.

**Pull request merged**